### DISCUSSION TITLE
How to - External Tasks and Import

### DISCUSSION THREAD

**Discussion by FLLApollo7** [pybricks/support#1971] [Q&A]
I can't find any documentation on the "External Task" option in Pybricks.  I think this is what I need to use to call a separate program that is saved under a separate file name.  The values to type into the open text boxes is not clear.  I'm also confused about how to call the new program and what the open circles represent there.  I tried to set up the overall program to call a separate program that included a task (attached below).  I get the error "invalid argument".  

I can find lots of examples with Python code to "import", but I need to do this with the blocks (not written out in code) as I'm trying to guide a team that is very new to this.  (I am very new too, and an initial attempt to "just type" in the right-hand code window said I was in read-only mode and couldn't edit. I code occasionally at work but only in very narrow programs, so there is much that I don't know.)  Another issue I'm confused by is one post calling "my_module" and another calling "our_robot", etc - what are these referring to and how are these "directories" (my mental categorization since I don't know a better term) set up?  

The overall goal is the ability to string together multiple programs in order for competition day.  I have figured out how to configure the button presses to do this and show feedback via the numbers on the hub - but only if we combine ALL the programs into the same file.  We historically have kept each "run" in a separate file, so I'm trying to figure out how to replicate that with this new programming method.  I need to understand this well enough to guide the kids through the logic so they can build it. 

![Overall_prog](https://github.com/user-attachments/assets/66ca38b4-1de1-42bc-9c46-3d7e83398d75)
![Prog1_mytask](https://github.com/user-attachments/assets/b39182ef-aa96-493e-977b-02b7a9cbf544)

**Comment by laurensvalk** [comment]
> The overall goal is the ability to string together multiple programs in order for competition day.

If this is the main goal, I recommend this approach: https://pybricks.com/project/spike-hub-menu/

You can make multiple totally separate bock programs (one for each mission), and put them together as shown above. The main menu is in Python, but you can copy the ready-made example.

**Reply by FLLApollo7** [reply]
Thanks. I have seen this - one of the "import" examples I mentioned. I still need help with: 
-What is hub_menu and why is that what I am calling? Is this a built in tool? How do I know that is what to import?
-How do I type in code? I tried on the right hand panel next to the block coding and it told me it was read only. I only downloaded Pybricks. I don't understand if other python coding software is needed for this? 
-Consider this a request for "a block" to do this as you asked in that link. The team needs to have a block option. 

I'd still like to understand the external task block and how to use it.

**Reply by laurensvalk** [reply]
Yes, it's a [builtin function](https://docs.pybricks.com/en/stable/tools/#pybricks.tools.hub_menu). The video demonstrates what it does.

> How do I type in code?

Create a new project and choose Python as the type. No other apps are needed.

(In block programs, the Python panel is read-only so you don't lose your changes as you move blocks around and generate new code.)

Consider starting by downloaded the simple examples in the link, before using it in your own programs.

> I'd still like to understand the external task block and how to use it.

Rather than being used for different missions, this is normally used to write a few functions that you want to re-use between projects. For example, you might make a module called 'my_module' with one or more functions such as `calculate_square`. Then other scripts can import this function and use it.
